pennylane/control_flow/for_loop.py

```diff
@@ -420,7 +420,10 @@ def _call_capture_enabled(self, *init_state):
 
     def __call__(self, *init_state):
 
-        if enabled():
-            return self._call_capture_enabled(*init_state)
+        bypass_condition = (
+            isinstance(self.stop, int) and isinstance(self.start, int) and self.stop == self.start
+        )
```

Comment:
or `has_zero_range`? 
```suggestion
        start_equals_stop = (
            isinstance(self.stop, int) and isinstance(self.start, int) and self.stop == self.start
        )
```
Easier to visually parse that way.